src/test/scala/com/microsoft/hyperspace/index/rules/RuleUtilsTest.scala

```diff
@@ -93,4 +94,24 @@ class RuleUtilsTest extends HyperspaceRuleTestSuite {
 
     assert(RuleUtils.getCandidateIndexes(indexManager, t1ProjectNode).length === 2)
   }
+
+  test("Verify get logical relation for single logical relation node plan.") {
+    validateLogicalRelation(t1ScanNode)
+  }
+
+  test("Verify get logical relation for multi-node linear plan.") {
+    validateLogicalRelation(t1ProjectNode)
+  }
+
+  test("Verify get logical relation for non-linear plan.") {
+    val joinNode = Join(t1ProjectNode, t2ProjectNode, JoinType("inner"), None)
+    val r = RuleUtils.getLogicalRelation(Project(Seq(t1c3, t2c3), joinNode))
+    assert(r.isEmpty)
+  }
+
+  private def validateLogicalRelation(plan: LogicalPlan): Unit = {
+    val r = RuleUtils.getLogicalRelation(plan)
+    assert(r.isDefined)
+    assert(r.get.equals(t1ScanNode))
```

Comment:
It should be `private def validateLogicalRelation(plan: LogicalPlan, expected: LogicalRelation)` and not tightly depend on `t1ScanNode`

Reply:
Fixed it.